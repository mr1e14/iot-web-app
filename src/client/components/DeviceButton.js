import React from "react";
import PropTypes from "prop-types";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import { Link } from "react-router-dom";

const DeviceButton = props => {
  const { className, text, iconSrc, iconAlt, linkTo } = props;

  return (
    <Button
      component={Link}
      to={linkTo}
      variant="contained"
      fullWidth={true}
      className={className}
    >
      <Typography color="textPrimary">{text}</Typography>
      <img src={iconSrc} alt={iconAlt} style={{ padding: "0.25rem 0.5rem" }} />
    </Button>
  );
};

DeviceButton.propTypes = {
  className: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  iconSrc: PropTypes.string.isRequired,
  linkTo: PropTypes.string.isRequired
};

export default DeviceButton;
