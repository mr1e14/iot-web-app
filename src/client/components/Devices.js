import React from "react";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import lightIcon from "../img/light.png";
import thermometerIcon from "../img/thermometer.png";
import humidityIcon from "../img/humidity.png";
import motionSensorIcon from "../img/motion-sensor.png";
import matches from "./mediaQuery";
import DeviceButton from "./DeviceButton";

const Devices = props => {
  const { classes } = props;

  return (
    <React.Fragment>
      <Grid className={classes.childGridCenter} item xs={12}>
        <Typography variant="h6">Smart devices</Typography>
      </Grid>
      <Grid
        className={
          matches("sm") ? classes.childGridLeft : classes.childGridCenter
        }
        item
        xs={12}
        sm={6}
      >
        <DeviceButton
          className={classes.deviceButton1}
          text="Manage lights"
          iconSrc={lightIcon}
          iconAlt="light-icon"
        />
      </Grid>
      <Grid
        className={
          matches("sm") ? classes.childGridRight : classes.childGridCenter
        }
        item
        xs={12}
        sm={6}
      >
        <DeviceButton
          className={classes.deviceButton3}
          text="5 minutes ago"
          iconSrc={motionSensorIcon}
          iconAlt="motion-sensor-icon"
        />
      </Grid>
      <Grid
        className={
          matches("sm") ? classes.childGridLeft : classes.childGridCenter
        }
        item
        xs={12}
        sm={6}
      >
        <DeviceButton
          className={classes.deviceButton2}
          text="House avg. 21Â°C"
          iconSrc={thermometerIcon}
          alt="thermometer-icon"
        />
      </Grid>
      <Grid
        className={
          matches("sm") ? classes.childGridRight : classes.childGridCenter
        }
        item
        xs={12}
        sm={6}
      >
        <DeviceButton
          className={classes.deviceButton4}
          text="House avg. 77%"
          iconSrc={humidityIcon}
          iconAlt="humidity-icon"
        />
      </Grid>
    </React.Fragment>
  );
};

export default Devices;
