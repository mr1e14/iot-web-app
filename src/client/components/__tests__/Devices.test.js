import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

jest.mock("../mediaQuery");

import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import DeviceButton from "../DeviceButton";
import Devices from "../Devices";
import matches from "../mediaQuery";

const exampleClasses = {
  child: "child",
  childGridCenter: "childGridCenter",
  childGridLeft: "childGridLeft",
  childGridRight: "childGridRight",
  button1: "button1",
  button2: "button2",
  button3: "button3",
  button4: "button4"
};

describe("Devices", () => {
  let tree;
  let component;
  describe("on xs devices", () => {
    beforeEach(() => {
      matches.mockReturnValue(true);
      component = renderer.create(<Devices classes={exampleClasses} />);
      tree = component.toJSON();
    });
    it("renders correctly", () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe("on larger devices", () => {
    beforeEach(() => {
      matches.mockReturnValue(false);
      component = renderer.create(<Devices classes={exampleClasses} />);
      tree = component.toJSON();
    });
    it("renders correctly", () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe("when renders", () => {
    let wrapper;
    describe("on xs devices", () => {
      beforeEach(() => {
        matches.mockReturnValue(true);
        wrapper = mount(<Devices classes={exampleClasses} />);
      });
      it("has a header with text in the centre", () => {
        expect(
          wrapper
            .find(Typography)
            .at(0)
            .text()
        ).toEqual("Smart devices");
        expect(
          wrapper
            .find(Grid)
            .at(0)
            .hasClass("childGridCenter")
        ).toBe(true);
      });
      it("has four DeviceButtons", () => {
        expect(wrapper.find(DeviceButton).length).toEqual(4);
      });
      it("has five Grids", () => {
        expect(wrapper.find(Grid).length).toEqual(5);
      });
      it("assigned child class to Grids with DeviceButton so there is one per line", () => {
        expect(
          wrapper
            .find(Grid)
            .at(1)
            .hasClass("child")
        ).toBe(true);
        expect(
          wrapper
            .find(Grid)
            .at(2)
            .hasClass("child")
        ).toBe(true);
        expect(
          wrapper
            .find(Grid)
            .at(3)
            .hasClass("child")
        ).toBe(true);
        expect(
          wrapper
            .find(Grid)
            .at(4)
            .hasClass("child")
        ).toBe(true);
      });
    });
    describe("on larger devices", () => {
      beforeEach(() => {
        matches.mockReturnValue(false);
        wrapper = mount(<Devices classes={exampleClasses} />);
      });
      it("has a header with text in the centre", () => {
        expect(
          wrapper
            .find(Typography)
            .at(0)
            .text()
        ).toEqual("Smart devices");
        expect(
          wrapper
            .find(Grid)
            .at(0)
            .hasClass("childGridCenter")
        ).toBe(true);
      });
      it("has four DeviceButtons", () => {
        expect(wrapper.find(DeviceButton).length).toEqual(4);
      });
      it("has five Grids", () => {
        expect(wrapper.find(Grid).length).toEqual(5);
      });
      it("assigned classes to Grids with DeviceButton such that there are two per line", () => {
        expect(
          wrapper
            .find(Grid)
            .at(1)
            .hasClass("childGridLeft")
        ).toBe(true);
        expect(
          wrapper
            .find(Grid)
            .at(2)
            .hasClass("childGridRight")
        ).toBe(true);
        expect(
          wrapper
            .find(Grid)
            .at(3)
            .hasClass("childGridLeft")
        ).toBe(true);
        expect(
          wrapper
            .find(Grid)
            .at(4)
            .hasClass("childGridRight")
        ).toBe(true);
      });
    });
  });
});
