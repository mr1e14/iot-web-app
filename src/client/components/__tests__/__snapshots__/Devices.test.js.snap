// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Devices on larger devices renders correctly 1`] = `
Array [
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 childGridCenter"
  >
    <h6
      className="MuiTypography-root-98 MuiTypography-h6-115"
    >
      Smart devices
    </h6>
  </div>,
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 childGridLeft"
  >
    <button
      className="MuiButtonBase-root-160 MuiButton-root-134 MuiButton-contained-145 MuiButton-raised-148 MuiButton-fullWidth-159 deviceButton1"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-135"
      >
        <p
          className="MuiTypography-root-98 MuiTypography-body1-107"
        >
          Manage lights
        </p>
        <img
          alt="light-icon"
          src="test-file-stub"
          style={
            Object {
              "padding": "0.25rem 0.5rem",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-163"
      />
    </button>
  </div>,
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 childGridRight"
  >
    <button
      className="MuiButtonBase-root-160 MuiButton-root-134 MuiButton-contained-145 MuiButton-raised-148 MuiButton-fullWidth-159 deviceButton3"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-135"
      >
        <p
          className="MuiTypography-root-98 MuiTypography-body1-107"
        >
          5 minutes ago
        </p>
        <img
          alt="motion-sensor-icon"
          src="test-file-stub"
          style={
            Object {
              "padding": "0.25rem 0.5rem",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-163"
      />
    </button>
  </div>,
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 childGridLeft"
  >
    <button
      className="MuiButtonBase-root-160 MuiButton-root-134 MuiButton-contained-145 MuiButton-raised-148 MuiButton-fullWidth-159 deviceButton2"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-135"
      >
        <p
          className="MuiTypography-root-98 MuiTypography-body1-107"
        >
          House avg. 21°C
        </p>
        <img
          src="test-file-stub"
          style={
            Object {
              "padding": "0.25rem 0.5rem",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-163"
      />
    </button>
  </div>,
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 childGridRight"
  >
    <button
      className="MuiButtonBase-root-160 MuiButton-root-134 MuiButton-contained-145 MuiButton-raised-148 MuiButton-fullWidth-159 deviceButton4"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-135"
      >
        <p
          className="MuiTypography-root-98 MuiTypography-body1-107"
        >
          House avg. 77%
        </p>
        <img
          alt="humidity-icon"
          src="test-file-stub"
          style={
            Object {
              "padding": "0.25rem 0.5rem",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-163"
      />
    </button>
  </div>,
]
`;

exports[`Devices on xs devices renders correctly 1`] = `
Array [
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 childGridCenter"
  >
    <h6
      className="MuiTypography-root-98 MuiTypography-h6-115"
    >
      Smart devices
    </h6>
  </div>,
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 childGridCenter"
  >
    <button
      className="MuiButtonBase-root-160 MuiButton-root-134 MuiButton-contained-145 MuiButton-raised-148 MuiButton-fullWidth-159 deviceButton1"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-135"
      >
        <p
          className="MuiTypography-root-98 MuiTypography-body1-107"
        >
          Manage lights
        </p>
        <img
          alt="light-icon"
          src="test-file-stub"
          style={
            Object {
              "padding": "0.25rem 0.5rem",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-163"
      />
    </button>
  </div>,
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 childGridCenter"
  >
    <button
      className="MuiButtonBase-root-160 MuiButton-root-134 MuiButton-contained-145 MuiButton-raised-148 MuiButton-fullWidth-159 deviceButton3"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-135"
      >
        <p
          className="MuiTypography-root-98 MuiTypography-body1-107"
        >
          5 minutes ago
        </p>
        <img
          alt="motion-sensor-icon"
          src="test-file-stub"
          style={
            Object {
              "padding": "0.25rem 0.5rem",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-163"
      />
    </button>
  </div>,
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 childGridCenter"
  >
    <button
      className="MuiButtonBase-root-160 MuiButton-root-134 MuiButton-contained-145 MuiButton-raised-148 MuiButton-fullWidth-159 deviceButton2"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-135"
      >
        <p
          className="MuiTypography-root-98 MuiTypography-body1-107"
        >
          House avg. 21°C
        </p>
        <img
          src="test-file-stub"
          style={
            Object {
              "padding": "0.25rem 0.5rem",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-163"
      />
    </button>
  </div>,
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 childGridCenter"
  >
    <button
      className="MuiButtonBase-root-160 MuiButton-root-134 MuiButton-contained-145 MuiButton-raised-148 MuiButton-fullWidth-159 deviceButton4"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-135"
      >
        <p
          className="MuiTypography-root-98 MuiTypography-body1-107"
        >
          House avg. 77%
        </p>
        <img
          alt="humidity-icon"
          src="test-file-stub"
          style={
            Object {
              "padding": "0.25rem 0.5rem",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-163"
      />
    </button>
  </div>,
]
`;
